lib/smart_on_fhir/epic.go

```diff
@@ -37,39 +35,37 @@ func NewEpicSmartOnFhirProvider(credentials *Credentials) *FhirProvider {
 	return &FhirProvider{
 		config: config,
 		UserInfo: func(t *FhirTokenResponse) (*UserInfo, error) {
-			// Steps:
-			// 1. Extract the sub from the access token
-			// 2. Combine the subject with the Practitioner endpoint to get user data
-			// 3. Unmarshal the response into a Practitioner object
-
-			var me struct {
-				id string
-				// telecom						contactPoint
+			keys, err := fetchEpicJWKS()
+			if err != nil {
+				return nil, err
 			}
 
-			oauth2Token := &oauth2.Token{
-				AccessToken: t.AccessToken,
-			}
+			idToken, err := jwt.ParseNoVerify([]byte(t.IdToken))
+			var epicIdTokenClaims *EpicIdTokenClaim
+			json.Unmarshal(idToken.Claims(), &epicIdTokenClaims)
 
-			client := config.Client(context.TODO(), oauth2Token)
-			resp, err := client.Get("https://fhir.epic.com/interconnect-fhir-oauth/api/FHIR/R4/Practitioner/")
-			if err != nil {
-				return nil, err
+			hasJwks := false
+			fmt.Println("==> Token kid:", idToken.Header().KeyID)
+			for index, key := range keys.Keys {
+				fmt.Println("===> Key", index + 1, "kid:", key.KeyID)
+
+				if key.KeyID == idToken.Header().KeyID {
+					hasJwks = true
+					break
+				}
 			}
-			defer resp.Body.Close()
 
-			body, err := ioutil.ReadAll(resp.Body)
-			if err != nil {
+			if hasJwks {
+				fmt.Println("=========== Key ID matches ===========")
```

Comment:
Are you planning on verifying the ID token using the key that we found in this PR or next?

Reply:
Which kind of verification do you want here? Isn't it just comparing the `kid`'s here?

Reply:
Converting the JWK into a format that the validate function can use is the hard part. A PEM might work but I'm not 100% sure exactly what format the Go JWT library needs